Azure arc servers sccm (#889)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_servers/scaled_deployment/configuration_manager_scaled_runscript/_index.md

---
type: docs
title: "Windows server with Configuration Manager using PowerShell"
linkTitle: "Windows server with Configuration Manager using PowerShell"
weight: 4
description: >
---
## Connect an existing Windows server to Azure Arc using Configuration Manager with PowerShell

> The following README will guide you on how to connect an Windows machine to Azure Arc with a PowerShell script using Configuration Manager.
> 
> This guide assumes that you already have an installation of Microsoft Configuration Manager, at least one active Windows server client and a basic understanding of the product.

Reviewer: ```suggestion
The following README will guide you on how to connect a Windows machine to Azure Arc with a PowerShell script using Configuration Manager.

This guide assumes that you already have an installation of [Microsoft Configuration Manager](https://docs.microsoft.com/en-us/mem/configmgr/core/understand/introduction), at least one active Windows server client, and a basic understanding of the product.
```

> ## Prerequisites

Reviewer: ```suggestion

## Prerequisites
```

> * Make sure you have all the Configuration Manager [pre-requisites](https://docs.microsoft.com/mem/configmgr/apps/deploy-use/create-deploy-scripts#prerequisites) in place to use the "Run Script" feature.
> * [Install or update Azure CLI to version 2.25.0 and above](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). Use the below command to check your current installed version.

Reviewer: ```suggestion
- Make sure you have met all the Configuration Manager [prerequisites](https://docs.microsoft.com/mem/configmgr/apps/deploy-use/create-deploy-scripts#prerequisites) in order to use the "Run Script" feature.

- [Install or update Azure CLI to version 2.25.0 and above](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). Use the below command to check your current installed version.
```

  ```shell
  az --version
  ```

* Enable subscription with the resource provider for Azure Arc-enabled Servers. Registration is an asynchronous process, and registration may take approximately 10 minutes.

  ```shell
  az provider register --namespace Microsoft.HybridCompute
  ```

  You can monitor the registration process with the following commands:

    ```shell
    az provider show -n Microsoft.HybridCompute -o table

>     ```

Reviewer: I think we need to consider adding this to the onboarding script

> * Create Azure service principal (SP)

Reviewer: ```suggestion
- Create Azure service principal (SP)
```

    ```shell
    az login
    az ad sp create-for-rbac -n "<Unique SP Name>" --role contributor
    ```

    For example:

    ```shell
    az ad sp create-for-rbac -n "http://AzureArcServers" --role contributor
    ```

    Output should look like this:

    ```json
    {
    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "displayName": "AzureArcServers",
    "name": "http://AzureArcServers",
    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
    ```

    > **Note: The Jumpstart scenarios are designed with as much ease of use in-mind and adhering to security-related best practices whenever possible. It is optional but highly recommended to scope the service principal to a specific [Azure subscription and resource group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest) as well considering using a [less privileged service principal account](https://docs.microsoft.com/en-us/azure/role-based-access-control/best-practices)**

> * Create a new Azure resource group where you want your machine(s) to show up.
> 
>     ![Screenshot showing Azure Portal with empty resource group](./01.png)
> 
> * Download the [az_connect_win](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/scripts/az_connect_win.ps1) PowerShell script.
> 
> * Change the environment variables according to your environment and save the script.

Reviewer: ```suggestion
- Create a new Azure resource group where you want your machine(s) to show up.

    ![Screenshot showing Azure Portal with empty resource group](./01.png)

- Download the [_az_connect_win_](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/scripts/az_connect_win.ps1) PowerShell script.

- Change the environment variables according to your environment and save the script.
```

    ![Screenshot showing PowerShell script](./02.png)

## Setting the PowerShell execution policy in Configuration Manager

In order for Configuration Manager to run the PowerShell script in this scenario, we will need to set the PowerShell execution policy and allow the script authors to approve their own scripts.

   > **Note: The Jumpstart scenarios are designed with as much ease of use in-mind and adhering to security-related best practices whenever possible. It is recommended to follow the security recommendations in your organization when it comes to PowerShell scripts execution policy and approvals**

> * Login to the Configuration Manager console.

Reviewer: ```suggestion
- Login to the Configuration Manager console.
```

would be good to add a screenshot here

> * After logging in, go to the “Administration” workspace. Under “Site Configuration”, select “Client Settings” and open the properties of the "Default Client Settings".

Reviewer: ```suggestion
- After logging in, go to the “Administration” workspace. Under “Site Configuration”, select “Client Settings” and open the properties of the "Default Client Settings".
```

   > **Note: You might have different client settings, choose the appropriate one that is deployed to the servers you are onboarding to Azure Arc**

>     ![Screenshot showing client settings in the configuration manager console](./03.png)

Reviewer: Adding steps number in the screenshot can help to better understand the screen flow

> * Click the “Computer Agent” settings, change the “PowerShell execution policy” to “Bypass”.

Reviewer: ```suggestion
- Click the “Computer Agent” settings, change the “PowerShell execution policy” to “Bypass”.
```

>     ![Screenshot showing Computer Agent settings in the configuration manager console](./04.png)

Reviewer: Adding steps numbers in the screenshot can help to better understand the screen flow

## Configuration of the PowerShell script approval settings in Configuration Manager

* Go to the “Administration” workspace. Under “Site Configuration”, select “Sites” and select the Configuration Manager site.
* Click on “Hierarchy Settings”.
    

>     ![Screenshot showing sites settings in the configuration manager console](./05.png)

Reviewer: Adding steps numbers in the screenshot can help to better understand the screen flow

> * In the “General” tab, uncheck the “Script authors require additional script approver” option.

Reviewer: ```suggestion
- In the “General” tab, uncheck the “Script authors require additional script approver” option.
```

    
    ![Screenshot showing script approval settings in the configuration manager console](./06.png)

## Deployment

> * Go to the “Software Library” workspace. Click on “Create Script”.

Reviewer: ```suggestion
- Go to the “Software Library” workspace. Click on “Create Script”.
```

>     ![Screenshot showing script creation in the configuration manager console](./07.png)

Reviewer: Adding steps numbers in the screenshot can help to better understand the screen flow

> * Provide a name to the script. Paste the content of the [az_connect_win](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/scripts/az_connect_win.ps1) PowerShell script you downloaded earlier. Click “Next” and “Close”.

Reviewer: ```suggestion
- Provide a name to the script. Paste the content of the [_az_connect_win_](https://github.com/microsoft/azure_arc/blob/main/azure_arc_servers_jumpstart/scripts/az_connect_win.ps1) PowerShell script you downloaded earlier. Click “Next” and “Close”.
```

    
    ![Screenshot showing script creation in the configuration manager console](./08.png)

> * Select the script and click “Approve/Deny”. Proceed to approve the script and close.

Reviewer: ```suggestion
- Select the script and click “Approve/Deny”. Proceed to approve the script and close.
```

    
    ![Screenshot showing script approval settings in the configuration manager console](./09.png)

    ![Screenshot showing script approving in the configuration manager console](./10.png)

> * Go to the “Assets and Compliance” workspace. Expand on “Device Collections” and select the collection that contains the sever(s) you want to onboard. Right-click the collection and click on “Run Script”.

Reviewer: ```suggestion
- Go to the “Assets and Compliance” workspace. Expand on “Device Collections” and select the collection that contains the server(s) you want to onboard. Right-click the collection and click on “Run Script”.
```

>     ![Screenshot showing opening the run script context menu in the configuration manager console](./11.png)

Reviewer: Adding steps numbers in the screenshot can help to better understand the screen flow

> * Select the script you created and click “Next”. Wait for the script to finish execution.

Reviewer: ```suggestion
- Select the script you created and click “Next”. Wait for the script to finish execution.
```

    ![Screenshot showing running the script in the configuration manager console](./12.png)

    ![Screenshot showing the script progress in the configuration manager console](./13.png)

>     ![Screenshot showing the script results in the configuration manager console](./14.png)

Reviewer: Might be good to add an arrow pointing to the red box since it's a bit hard to catch

> * Click on the “Run Details” tab. Double click on “Script Output” to see the output of the script.

Reviewer: ```suggestion
- Click on the “Run Details” tab. Double click on “Script Output” to see the output of the script.
```

    ![Screenshot showing the script result message in the configuration manager console](./15.png)

    ![Screenshot showing the script output in the configuration manager console](./16.png)

> * Upon completion, you will have your Windows server, connected as a new Azure Arc resource inside your resource group.

Reviewer: ```suggestion
- Upon completion, you will have your Windows server, connected as a new Azure Arc-enabled server resource inside your resource group.
```
